src/contentmanager.cpp

```diff
@@ -906,5 +906,6 @@ void ContentManager::updateLibraryFromDir(QString monitorDir)
 
 void ContentManager::handleDisappearedZimFile(QString bookId)
 {
-    mp_library->removeBookFromLibraryById(bookId);
+    if (!KiwixApp::instance()->getTabWidget()->getTabZimIds().contains(bookId))
+        mp_library->removeBookFromLibraryById(bookId);
```

Comment:
@veloman-yunkan I checked that the book's isPathValid boolean was already set when we retrieved it. Don't think we need to make any modifications to the library.